packages/app/src/cli/services/environment/id-matching.ts

```diff
@@ -1,137 +1,159 @@
 }
 
-export interface LocalExtension {
-  localIdentifier: string
-  graphQLType: string
-  type: ExtensionTypes
-  configuration: {name: string}
+/**
+ * Filter function to match a local and a remote source by type and name
+ */
+const sameTypeAndName = (local: LocalSource, remote: RemoteSource) => {
+  return remote.type === local.graphQLType && string.slugify(remote.title) === string.slugify(local.configuration.name)
 }
 
-export async function automaticMatchmaking(
-  localExtensions: LocalExtension[],
-  remoteRegistrations: ExtensionRegistration[],
-  identifiers: {[localIdentifier: string]: string},
-  registrationIdField: 'id' | 'uuid',
-): Promise<Result<MatchResult, MatchingError>> {
-  if (remoteRegistrations.length > localExtensions.length) {
-    return err('invalid-environment')
-  }
-
-  const validIdentifiers = identifiers
-
-  // Get the local UUID of an extension, if exists
-  const localId = (extension: LocalExtension) => validIdentifiers[extension.localIdentifier]
+/**
+ * Automatically match local and remote sources if they have the same type and name
+ *
+ * If multiple local or remote sources have the same type and name, they can't be matched automatically
+ */
+function matchByNameAndType(
+  local: LocalSource[],
+  remote: RemoteSource[],
+  remoteIdField: 'id' | 'uuid',
+): {matched: IdentifiersExtensions; pending: {local: LocalSource[]; remote: RemoteSource[]}} {
+  const uniqueLocal = uniqBy(local, (elem) => [elem.graphQLType, elem.configuration.name])
+  const uniqueRemote = uniqBy(remote, (elem) => [elem.type, elem.title])
+  const validMatches: IdentifiersExtensions = {}
+
+  uniqueLocal.forEach((localSource) => {
+    const possibleMatch = uniqueRemote.find((remoteSource) => sameTypeAndName(localSource, remoteSource))
+    if (possibleMatch) validMatches[localSource.localIdentifier] = possibleMatch[remoteIdField]
+  })
 
-  // All local UUIDs available
-  const localUUIDs = () => Object.values(validIdentifiers)
+  const pendingLocal = local.filter((elem) => !validMatches[elem.localIdentifier])
+  const pendingRemote = remote.filter(
+    (registration) => !Object.values(validMatches).includes(registration[remoteIdField]),
+  )
+  return {matched: validMatches, pending: {local: pendingLocal, remote: pendingRemote}}
+}
 
-  // Whether an extension has an UUID and that UUID and type match with a remote extension
-  const existsRemotely = (extension: LocalExtension) => {
-    const remote = remoteRegistrations.find((registration) => registration[registrationIdField] === localId(extension))
-    return remote !== undefined && remote.type === extension.graphQLType
+/**
+ * Ask the user to confirm the relationship between a local source and a remote source if they
+ * the only ones of their types.
+ */
+function matchByUniqueType(
+  localSources: LocalSource[],
+  remoteSources: RemoteSource[],
+): {
+  toCreate: LocalSource[]
+  toConfirm: {local: LocalSource; remote: RemoteSource}[]
```

Comment:
Just adding a note that this file also would reuse the same interface defined above.